details.py

```diff
@@ -0,0 +1,20 @@
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from datetime import datetime, timezone
+from catalog import models as cmod
+import math
+
+@view_function
+def process_request(request, product:cmod.Product):
+    categories = cmod.Category.objects.all()
+
+    if product in request.last_five:
+        request.last_five.remove(product)
+    request.last_five.insert(0, product)
+
+    if len(request.last_five) > 6:
```

Comment:
I did an insert here instead of a pop, I don't know if one is better than the other but that is another way to do it.

Reply:
Probably a good idea, just popping it off can be dangerous